extensions/data-plane/data-plane-google-storage/src/main/java/org/eclipse/edc/connector/dataplane/gcp/storage/DataPlaneGcsExtension.java

```diff
@@ -49,11 +50,12 @@ public String name() {
     public void initialize(ServiceExtensionContext context) {
 
         var monitor = context.getMonitor();
+        var gcpCredential = new GcpCredentials(vault, typeManager, monitor);
```

Comment:
Does it make sense to have GcpCredentials taking also the GcpConfiguration object injected?
If a service account is provided in the dataAddress, then it overrides the GcpConfiguration.